src/Analyzers/Core/CodeFixes/ImplementInterface/ImplementInterfaceGenerator.cs

```diff
@@ -405,22 +405,35 @@ private bool HasMatchingMember(ArrayBuilder<ISymbol> implementedVisibleMembers,
             return implementedVisibleMembers.Any(m => MembersMatch(m, member));
         }
 
-        private bool MembersMatch(ISymbol member1, ISymbol member2)
+        private bool MembersMatch(ISymbol existingMember, ISymbol memberToAdd)
         {
-            if (member1.Kind != member2.Kind)
+            if (existingMember.Kind != memberToAdd.Kind)
                 return false;
 
-            if (member1.DeclaredAccessibility != member2.DeclaredAccessibility ||
-                member1.IsStatic != member2.IsStatic)
+            if (existingMember.DeclaredAccessibility != memberToAdd.DeclaredAccessibility ||
+                existingMember.IsStatic != memberToAdd.IsStatic)
             {
                 return false;
             }
 
-            if (member1.ExplicitInterfaceImplementations().Any() || member2.ExplicitInterfaceImplementations().Any())
+            if (existingMember.ExplicitInterfaceImplementations().Any() || memberToAdd.ExplicitInterfaceImplementations().Any())
                 return false;
 
-            return SignatureComparer.Instance.HaveSameSignatureAndConstraintsAndReturnTypeAndAccessors(
-                member1, member2, IsCaseSensitive);
+            if (!SignatureComparer.Instance.HaveSameSignatureAndConstraintsAndReturnType(existingMember, memberToAdd, IsCaseSensitive))
+                return false;
+
+            if (existingMember is IPropertySymbol existingProperty && memberToAdd is IPropertySymbol propertyToAdd)
```

Comment:
>if (existingMember is IPropertySymbol existingProperty && memberToAdd is IPropertySymbol propertyToAdd) [](http://example.com/codeflow?start=12&length=103)

Coffee still isn't hitting, doesn't SignatureComparer.HaveSameSignatureAndConstraintsAndReturnType now cover this case?

Reply:
Discussed offline